lib/PlainSerializer.js

```diff
@@ -24,12 +24,29 @@ class PlainSerializer {
       '}'
     ].join('\n')
 
+    let leadingLines
+    switch (module) {
+      case 'esm':
+        leadingLines = ['export const dataset = ({ blankNode, literal, namedNode, quad }) => {']
+        break
+      case 'ts':
+        leadingLines = [
+          "import { DataFactory, Quad } from 'rdf-js'",
+          '',
+          'export const dataset = ({ blankNode, literal, namedNode, quad }: DataFactory): Quad[] => {'
+        ]
+        break
+      default:
+        leadingLines = ['module.exports = ({ blankNode, literal, namedNode, quad }) => {']
+        break
+    }
+
```

Comment:
Maybe it makes sense to use subclasses to handle this? E.g. TS may also use a different code style.

Reply:
Yea, why not 👍

Reply:
Instead of going for subclassing I went for a strategy pattern. For now only the `leadingLines` but could extract other portions in the future which shall differ between output types